simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -938,7 +938,14 @@ class simd<std::int32_t, simd_abi::avx2_fixed_size<4>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
                                                        element_aligned_tag) {
+    // FIXME_HIP ROCm 5.6 can't compile with the intrinsic used here.
+#if defined(__HIPCC__) &&       \
+    ((HIP_VERSION_MAJOR > 5) || \
+     ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR >= 6)))
```

Comment:
```suggestion
     ((HIP_VERSION_MAJOR >= 5) && (HIP_VERSION_MINOR >= 6)))
```
Don't we want to only guard against rocm 5.6+ and not necessarily for everything after rocm 5.0?

Reply:
What you are proposing wouldn't use the workaround for, say, 6.0, right? I think the current version is correct for guarding >=5.6. For 5.0 we wouldn't use the workaround since both `HIP_VERSION_MAJOR > 5` and `HIP_VERSION_MINOR >=6` are false.